src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -0,0 +1,1296 @@
+            string source = @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        /*<bind>*/int i = GetInt(), j = GetInt()/*</bind>*/;
+    }
+
+    static int GetInt() => 1;
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (2 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: System.Int32 i (OperationKind.VariableDeclaration)
+    Initializer: IInvocationExpression (static System.Int32 Program.GetInt()) (OperationKind.InvocationExpression, Type: System.Int32)
+  IVariableDeclaration: System.Int32 j (OperationKind.VariableDeclaration)
+    Initializer: IInvocationExpression (static System.Int32 Program.GetInt()) (OperationKind.InvocationExpression, Type: System.Int32)
+";
+            VerifyOperationTreeForTest<VariableDeclarationSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact]
+        public void SingleVariableDeclarationLocalReferenceInitializer()
+        {
+            string source = @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        int i = 1;
+        /*<bind>*/int i1 = i/*</bind>*/;
+    }
+
+    static int GetInt() => 1;
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: System.Int32 i1 (OperationKind.VariableDeclaration)
+    Initializer: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+";
+            VerifyOperationTreeForTest<VariableDeclarationSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact]
+        public void MultipleDeclarationsLocalReferenceInitializers()
+        {
+            string source = @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        int i = 1;
+        /*<bind>*/int i1 = i, i2 = i1/*</bind>*/;
+    }
+
+    static int GetInt() => 1;
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (2 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: System.Int32 i1 (OperationKind.VariableDeclaration)
+    Initializer: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+  IVariableDeclaration: System.Int32 i2 (OperationKind.VariableDeclaration)
+    Initializer: ILocalReferenceExpression: i1 (OperationKind.LocalReferenceExpression, Type: System.Int32)
+";
+            VerifyOperationTreeForTest<VariableDeclarationSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact]
+        public void InvalidArrayDeclaration()
+        {
+            string source = @"
+class Program
+{
+    static void Main(string[] args)
+    {
+        /*<bind>*/int[2, 3] a/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement)
+  IVariableDeclaration: System.Int32[,] a (OperationKind.VariableDeclaration)
```

Comment:
Agreed, I opened #18351. I think that it's related to the IInvalidExpression issue, where that is actually just another instance of the fact that nodes can hide invalid syntax under them when there is no place for it to go. In this case, there isn't even any information that there _are_ invalid nodes in the tree from here, even though said nodes can be retrieved.